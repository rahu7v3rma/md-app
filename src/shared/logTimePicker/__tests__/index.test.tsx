import moment from 'moment';
import React from 'react';
import renderer, { act } from 'react-test-renderer';

import LogTimePicker from '@/shared/logTimePicker';

it('logTimePicker snapshot', async () => {
    const selectedValue = new Date(`2023-01-01 12:00:00`);
    const tree = renderer.create(
        <LogTimePicker
            fieldName="Start Time"
            selectedValue={selectedValue}
            locale="en_GB"
        />
    );
    expect(tree).toMatchSnapshot();
});

it('logTimePicker renders without crashing', async () => {
    const selectedValue = new Date(`2023-01-01 12:00:00`);
    const tree = renderer.create(
        <LogTimePicker
            fieldName="Start Time"
            selectedValue={selectedValue}
            locale="en_GB"
        />
    );
    expect(tree.root).toBeTruthy();
});

it('logTimePicker displays the selected time correctly', async () => {
    const selectedValue = new Date(`2023-01-01 12:00:00`);
    const tree = renderer.create(
        <LogTimePicker
            fieldName="Start Time"
            selectedValue={selectedValue}
            locale="en_GB"
        />
    );
    const valueTxt = tree.root.findByProps({
        testID: 'valueTxt'
    }).props;
    expect(valueTxt.children).toEqual(moment(selectedValue).format('HH:mm'));
});

it('logTimePicker calls onSelect callback when time is selected', async () => {
    const selectedValue = new Date(`2023-01-01 12:00:00`);
    const onSelect = jest.fn();

    const tree = renderer.create(
        <LogTimePicker
            fieldName="Start Time"
            selectedValue={selectedValue}
            onSelect={onSelect}
            locale="en_GB"
        />
    );

    const logInputContainer = tree.root.findByProps({
        testID: 'logInputContainer'
    }).props;
    await act(() => logInputContainer.onPress());

    const saveBtn = tree.root.findByProps({
        testID: 'saveBtn'
    }).props;
    await act(() => saveBtn.onPress());

    expect(onSelect).toHaveBeenCalledTimes(1);
});

it('logTimePicker displays custom title when provided', async () => {
    const selectedValue = new Date(`2023-01-01 12:00:00`);
    const customTitle = '12:00:00';

    const tree = renderer.create(
        <LogTimePicker
            fieldName="Start Time"
            selectedValue={selectedValue}
            customTitle={customTitle}
            locale="en_GB"
        />
    );
    const valueTxt = tree.root.findByProps({
        testID: 'valueTxt'
    }).props;
    expect(valueTxt.children).toEqual(customTitle);
});

it('logTimePicker disables interaction when disabled prop is set', async () => {
    const selectedValue = new Date(`2023-01-01 12:00:00`);
    const disabled = true;

    const tree = renderer.create(
        <LogTimePicker
            fieldName="Start Time"
            selectedValue={selectedValue}
            locale="en_GB"
            disabled={disabled}
        />
    );

    // Open time picker
    const logInputContainer = tree.root.findByProps({
        testID: 'logInputContainer'
    }).props;
    await act(() => logInputContainer.onPress());

    let timerPickerContainer;
    try {
        timerPickerContainer = tree.root.findByProps({
            testID: 'timerPickerContainer'
        });
    } catch (error) {}
    expect(timerPickerContainer).toBeUndefined();
});
