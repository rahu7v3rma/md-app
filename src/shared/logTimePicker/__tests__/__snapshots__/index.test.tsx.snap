// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`logTimePicker snapshot 1`] = `
[
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#FFFFFF",
        "borderRadius": 16,
        "elevation": 5,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginVertical": 8,
        "opacity": 1,
        "paddingHorizontal": 20,
        "paddingVertical": 18,
        "shadowColor": "rgba(0,0,0,0.5)",
        "shadowOffset": {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 16,
      }
    }
    testID="logInputContainer"
  >
    <Text
      adjustsFontSizeToFit={true}
      style={
        [
          {
            "fontFamily": "Poppins",
          },
          {
            "flex": 1,
            "lineHeight": 20,
          },
          {
            "color": "#58595A",
            "fontSize": 14,
            "fontWeight": "500",
            "lineHeight": 22,
          },
        ]
      }
      testID="fieldNameTxt"
    >
      Start Time
    </Text>
    <View
      style={
        {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <Text
        adjustsFontSizeToFit={true}
        style={
          [
            {
              "fontFamily": "Poppins",
            },
            {
              "flex": 1,
              "lineHeight": 20,
              "paddingRight": 10,
              "textAlign": "right",
            },
            {
              "color": "#271A51",
              "fontSize": 14,
              "fontWeight": "600",
              "lineHeight": 22,
            },
          ]
        }
        testID="valueTxt"
      >
        12:00
      </Text>
    </View>
  </View>,
  <Modal
    animationType="none"
    deviceHeight={null}
    deviceWidth={null}
    hardwareAccelerated={false}
    hideModalContentWhileAnimating={false}
    onBackdropPress={[Function]}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    onRequestClose={[Function]}
    onSwipeComplete={[Function]}
    panResponderThreshold={4}
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    statusBarTranslucent={false}
    supportedOrientations={
      [
        "portrait",
        "landscape",
      ]
    }
    swipeDirection="down"
    swipeThreshold={100}
    testID="timePickerModal"
    transparent={true}
    visible={false}
  />,
]
`;
